### PR TITLE
New architecture with setuptools

### PR DESCRIPTION
**Pull request by k4nar**
This PR changes the architecture of the repository in order to facilitate Onitu's packaging. It also adds MIT license as requested in #18 .

In a nutshell, each driver is now a separate package which will be pushed on PyPI. Onitu is also a package, and will get the installed drivers using the `onitu.drivers` entry point.

Now onitu should be installed by executing `pip install -r requirements.txt`, which will install all the required dependencies for development and testing, and export the `onitu` command which should be used instead of the old `python -m onitu`.

By default Onitu doesn't come with any driver as I wasn't able to find a solution to tell that it depends on packages available in the repo. If you want to install a driver, you can do `pip install -e drivers/driver_name`. If you installed onitu with the `requirements.txt` file, Local Storage is already installed, but not if you just did `pip install -e .`.
We could add the official drivers to Onitu's dependency, but right now it wouldn't work as the packages are not on PyPI.

The package is almost ready to be pushed to PyPI, but I didn't look at the wheels and some other packaging issues. This is something we should check someday.

### CODE REVIEW

File: requirements_docs.txt [hunk 1]
```diff
@@ -1,2 +0,0 @@
-sphinx
-sphinxcontrib-httpdomain
```
Comment (Khady) [inline requirements_docs.txt:2]: We still need `sphinxcontrib-httpdomain` to build the doc. We should keep this file or put `sphinxcontrib-httpdomain` somewhere in the `setup.py`.

**Comment by Khady** [comment]
I think we should add an AUTHORS or CONTRIBUTORS file with the name of each member of the team.

File: requirements_docs.txt (hunk 1 above)
Comment (k4nar) [inline requirements_docs.txt:2]: Yep, sorry for that, I'll fix it.

**Comment by k4nar** [comment]
Yep, I'll do that. Usually in the CONTRIBUTORS file we put an email address for each contributor. So we should ask each member of the group which address they would like to use.

**Comment by k4nar** [comment]
We'll add a CONTRIBUTORS file later. I merge it in order to avoid any conflict.

**Pull request merged**